tests/test_test_utils.py

```diff
@@ -58,7 +69,39 @@ def test_get_route():
                 resp = yield from client.request("GET", "/")
                 assert resp.status == 200
                 text = yield from resp.text()
-                assert "Hello, world" in text
+                assert _hello_world_str == text
+
+            loop.run_until_complete(test_get_route())
+
+
+def test_auto_gzip_decompress():
+    with loop_context() as loop:
+        app = _create_example_app()
+        with _TestClient(_TestServer(app, loop=loop), loop=loop) as client:
+
+            @asyncio.coroutine
+            def test_get_route():
+                nonlocal client
+                resp = yield from client.request("GET", "/gzip_hello")
+                assert resp.status == 200
+                data = yield from resp.read()
+                assert data == _hello_world_bytes
+
+            loop.run_until_complete(test_get_route())
+
+
+def test_noauto_gzip_decompress():
+    with loop_context() as loop:
+        app = _create_example_app()
+        with _TestClient(_TestServer(app, loop=loop), auto_decompress=False) as client:
+
+            @asyncio.coroutine
+            def test_get_route():
+                nonlocal client
+                resp = yield from client.request("GET", "/gzip_hello")
+                assert resp.status == 200
+                data = yield from resp.read()
+                assert data == _hello_world_gz
```

Comment:
I've been avoiding too much refactoring because @asvetlov has frowned upon that in the past. He's wanted these types of changes in multiple PRs. If he gives the go-ahead I can do these suggested changes.